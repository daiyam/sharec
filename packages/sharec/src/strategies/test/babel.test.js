const { babelStrategy } = require('strategies/babel')

describe('strategy > babel', () => {
  describe('json strategy', () => {
    // Base merge case
    const babel01 = require('fixtures/babel/json/babel_01.json')
    const babel02 = require('fixtures/babel/json/babel_02.json')

    // Env merging features
    const babel04 = require('fixtures/babel/json/babel_04.json')
    const babel05 = require('fixtures/babel/json/babel_05.json')
    const babel07 = require('fixtures/babel/json/babel_07.json')
    const babel08 = require('fixtures/babel/json/babel_08.json')

    // Uapply features
    const babel10 = require('fixtures/babel/json/babel_10.json')
    const babel11 = require('fixtures/babel/json/babel_11.json')

    it('should merge babel json configs', () => {
      expect(babelStrategy.mergeJSON(babel01, babel02)).toMatchSnapshot()
    })

    it('should handle unique env cases during merging', () => {
      expect(
        babelStrategy.merge('.babelrc')(babel04, babel05),
      ).toMatchSnapshot()
      expect(
        babelStrategy.merge('.babelrc.json')(babel05, babel04),
      ).toMatchSnapshot()
      expect(
        babelStrategy.merge('.babelrc')(babel07, babel08),
      ).toMatchSnapshot()
      expect(babelStrategy.merge('babel')(babel08, babel07)).toMatchSnapshot()
    })

    it('should unapply babel JSON config', () => {
      expect(babelStrategy.unapplyJSON(babel10, babel11)).toMatchSnapshot()
    })

    it('should unapply babel config by file', () => {
      expect(
        babelStrategy.unapply('.babelrc')(babel11, babel10),
      ).toMatchSnapshot()
    })
  })
})
