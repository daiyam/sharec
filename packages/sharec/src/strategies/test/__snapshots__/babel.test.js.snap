// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strategy > babel json strategy should handle unique env cases during merging 1`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "env",
      ],
    },
  },
}
`;

exports[`strategy > babel json strategy should handle unique env cases during merging 2`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "env",
      ],
    },
  },
}
`;

exports[`strategy > babel json strategy should handle unique env cases during merging 3`] = `
Object {
  "env": Object {
    "development": Object {
      "presets": Array [
        "stage-2",
      ],
    },
    "test": Object {
      "presets": Array [
        "env",
      ],
    },
  },
}
`;

exports[`strategy > babel json strategy should handle unique env cases during merging 4`] = `
Object {
  "env": Object {
    "development": Object {
      "presets": Array [
        "stage-2",
      ],
    },
    "test": Object {
      "presets": Array [
        "env",
      ],
    },
  },
}
`;

exports[`strategy > babel json strategy should merge babel json configs 1`] = `
Object {
  "env": Object {
    "test": Object {
      "plugins": Array [
        "dynamic-import-node",
      ],
      "presets": Array [
        "env",
        "stage-2",
      ],
    },
  },
  "ignore": Array [
    "*.babelrc",
  ],
  "plugins": Array [
    Array [
      "syntax-dynamic-import",
      Object {
        "foo": "bar",
      },
    ],
    "lodash",
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "last 2 versions",
            "not ie > 0",
          ],
        },
      },
    ],
    "stage-2",
  ],
}
`;

exports[`strategy > babel json strategy should unapply babel JSON config 1`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "stage-2",
      ],
    },
  },
  "plugins": Array [
    Array [
      "syntax-dynamic-import",
      Object {
        "foo": "bar",
      },
    ],
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "last 2 versions",
            "not ie > 10",
            "not ie > 0",
          ],
        },
      },
    ],
  ],
}
`;

exports[`strategy > babel json strategy should unapply babel config by file 1`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "stage-2",
      ],
    },
  },
  "plugins": Array [
    Array [
      "syntax-dynamic-import",
      Object {
        "foo": "bar",
      },
    ],
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "last 2 versions",
            "not ie > 10",
            "not ie > 0",
          ],
        },
      },
    ],
  ],
}
`;
